### ISSUE TITLE
Running `az functionapp identity assign` causes error `'NoneType' object has no attribute 'identity'`

### ISSUE THREAD

**Issue by roodee** [MicrosoftDocs/azure-dev-docs#727]
Hello everyone, 

Running the command 

```bash
# Generate managed service identity for function app
az functionapp identity assign \
    --resource-group $resource_group_name \
    --name $funcapp_name
```
as described in section "7. Assign access policies and roles" of [Create resources for a cloud-based, serverless ETL solution using Python on Azure | Microsoft Docs](https://docs.microsoft.com/en-us/azure/developer/python/tutorial-deploy-serverless-cloud-etl-02#7-assign-access-policies-and-roles) causes the following error:

```bash
The command failed with an unexpected error. Here is the traceback:
'NoneType' object has no attribute 'identity'
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 658, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 721, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 692, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 971, in assign_identity
    webapp = _assign_identity(cmd.cli_ctx, getter, setter, role, scope)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/arm.py", line 718, in assign_identity
    resource = setter(resource)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 943, in setter
    if webapp.identity and webapp.identity.type == ResourceIdentityType.system_assigned_user_assigned:
AttributeError: 'NoneType' object has no attribute 'identity'
To open an issue, please run: 'az feedback'
```

Both variables are set as

```bash
funcapp_name='CloudETLFunc'
resource_group_name='rg_cloudetl_demo'
```

I am running the code on Azure's Cloud Shell. The output of  `az --version`:

```bash
azure-cli                         2.34.0 *

core                              2.34.0 *
telemetry                          1.0.6

Extensions:
ai-examples                        0.2.5
ssh                                1.0.0

Dependencies:
msal                              1.16.0
azure-mgmt-resource               20.0.0

Python location '/opt/az/bin/python3'
Extensions directory '/home/USERNAME/.azure/cliextensions'
Extensions system directory '/opt/az/lib/python3.8/site-packages/azure-cli-extensions'

Python (Linux) 3.8.12 (default, Feb 25 2022, 06:05:06) 
[GCC 8.3.0]
```

Do I run into this error due to missing permission? Or do I have to use an old version of Azure CLI?  

The tutorial seems very promising, what its usability as template for own ETL solutions is concerned. Unfortunately the promise is currently not kept. Thank you for your support.

Kind regards

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3bbb5dd1-f7fa-5ebe-1c47-1eef3182db5b
* Version Independent ID: 97bc93df-959a-ffd6-a084-ffeb99a2d94c
* Content: [Create resources for a cloud-based, serverless ETL solution using Python on Azure](https://docs.microsoft.com/en-us/azure/developer/python/tutorial-deploy-serverless-cloud-etl-02)
* Content Source: [articles/python/tutorial-deploy-serverless-cloud-etl-02.md](https://github.com/MicrosoftDocs/azure-dev-docs/blob/main/articles/python/tutorial-deploy-serverless-cloud-etl-02.md)
* Product: **azure-python**
* GitHub Login: @jess-johnson-msft
* Microsoft Alias: **jejohn**

**Comment by jess-johnson-msft** [comment]
Thanks for the feedback - I will start to triage the issue and work on a resolution!

**Comment by dbradish-microsoft** [comment]
@diberry , Required minimum Azure CLI version is exposed on the [Available Azure CLI extensions](https://learn.microsoft.com/en-us/cli/azure/azure-cli-extensions-list?view=azure-cli-latest) page, but not for Core commands.  `az functionapp` is an extension and requires Azure CLI version **2.0.46**.  Please keep in mind that as a reference group matures, parameters are added that may require a higher version. 

Other items that may be of interest:
* This command group is owned by the Azure Functions engineering team and not by the Azure CLI engineering team.
* The reference documentation instructions the user to install Azure Functions Core Tools by following [these GitHub instructions](https://github.com/Azure/azure-functions-core-tools).  This is not the standard way to install an Azure CLI extension.

**Comment by diberry** [comment]
@roodee It appears you might not have the azure-functions-ore-tools installed. Can you verify that by running the following command and posting of the results include `azure-functions-core-tools` and if it does, what version? 

```
npm list -g --depth=0
```

**Comment by roodee** [comment]
@diberry: Thank you very much for your support. In the mean time I chose not to use `az` but the Python SDK instead. It worked fine. Thanks again.

**Comment by diberry** [comment]
@roodee Can you tell me more about that? Which Python SKD?

**Comment by roodee** [comment]
@diberry : The project I am working in is about implementing an ETL process for binary files, extract metadata to enable a later search and at the end move the files and the metadata files into a blob storage. In the beginning I was curious about using Azure functions and following the description I linked above: https://docs.microsoft.com/en-us/azure/developer/python/tutorial-deploy-serverless-cloud-etl-02#7-assign-access-policies-and-roles.

To answer your question I copied some lines out of our requirements-file. We implemented the whole process in plain Python also using Azure-libraries for Python in the versions as follows:
```
azure-common==1.1.28
azure-core==1.25.1
azure-identity==1.10.0
azure-keyvault-secrets==4.5.1
```

I hope this answer is what you wanted to know.
